src/Interpreters/GraceHashJoin.cpp

```diff
@@ -168,20 +163,17 @@ class GraceHashJoin::FileBucket : boost::noncopyable
 
     bool empty() const { return is_empty.load(); }
 
-    Stats getStat() const { return stats; }
-
     AccumulatedBlockReader startJoining()
     {
         LOG_TRACE(log, "Joining file bucket {}", idx);
 
-        {
```

Comment:
I'd rather leave the scope. But the scope is related to file access mutex, not to `state`